src/Linters/PreferLambdasLinter.hack

```diff
@@ -82,4 +82,24 @@ final class PreferLambdasLinter extends AutoFixingASTLinter {
       $body,
     );
   }
+
+  private function simplifyBody(ILambdaBody $body): ILambdaBody {
```

Comment:
If I'm reading everything correctly, `$body` here should be guaranteed to be `CompoundStatement` (because it's the result of calling `getBody()` on an `AnonymousFunction` object above). If you update the type annotation, that could simplify the code below (you can get the statements directly without the initial checks):

```hack
$statements = $body->getStatements()?->getChildren();
```

(or something like that)

Reply:
Here is what I have after the update
```
  private function simplifyBody(CompoundStatement $body): ILambdaBody {
    $statements = $body->getStatements()?->getChildren();
    if (!$statements || C\count($statements) !== 1) {
      // Body has no statements, cannot be simplied.
      return $body;
    }
    $statement = C\onlyx($statements);
    if (!$statement is ReturnStatement) {
      // Only a return statement can be simplified.
      return $body;
    }
    return $statement->getExpressionx();
  }
```

Even though the code works, the return type isn't really as expected since LambdaExpression object is expecting an ILambdaBody as the body but we are returning an IExpression. I tried searching through existing classes that implements ILambdaBody but can't seem to find a good match to use, do we need to create a new class to handle that? (maybe a SingularStatement class vs CompountStatement)

Reply:
> LambdaExpression object is expecting an ILambdaBody as the body but we are returning an IExpression

Good catch, although it's weird why the same problem didn't affect the original version of the code.

I think you could just add `extends ILambdaBody` to the `IExpression` interface declaration -- as far as I can tell it's just an oversight that it isn't declared as such already.